lib/base/utility.cpp

```diff
@@ -1533,28 +1535,58 @@ static String UnameHelper(char type)
 	}
 }
 #endif /* _WIN32 */
+
+#ifdef _WIN32
+struct RegistryString
+{
+	union {
+		BYTE AsBytes[512];
+		char AsChars[1] = {0};
+	} Data;
+
+	DWORD Size = sizeof(Data);
+};
+
+static const char * const l_RegCurrentVersion = R"EOF(SOFTWARE\Microsoft\Windows NT\CurrentVersion)EOF";
+#endif /* _WIN32 */
+
 static bool ReleaseHelper(String *platformName, String *platformVersion)
 {
 #ifdef _WIN32
 	if (platformName)
 		*platformName = "Windows";
 
 	if (platformVersion) {
-		*platformVersion = "Vista";
-		if (IsWindowsVistaSP1OrGreater())
-			*platformVersion = "Vista SP1";
-		if (IsWindowsVistaSP2OrGreater())
-			*platformVersion = "Vista SP2";
-		if (IsWindows7OrGreater())
-			*platformVersion = "7";
-		if (IsWindows7SP1OrGreater())
-			*platformVersion = "7 SP1";
-		if (IsWindows8OrGreater())
-			*platformVersion = "8";
-		if (IsWindows8Point1OrGreater())
-			*platformVersion = "8.1 or greater";
-		if (IsWindowsServer())
-			*platformVersion += " (Server)";
+		HKEY hKey;
+		auto err (RegOpenKeyExA(HKEY_LOCAL_MACHINE, l_RegCurrentVersion, 0, KEY_READ, &hKey));
+
+		if (err == ERROR_SUCCESS) {
+			Defer regCloseKey ([hKey]() { (void)RegCloseKey(hKey); });
+			RegistryString productName;
+			auto err (RegQueryValueExA(hKey, "ProductName", nullptr, nullptr, productName.Data.AsBytes, &productName.Size));
```

Comment:
Maybe you need another subkey to search for instead of `ProductName` e.g `OsName`, otherwise, it's still printing false positive information (Tested on fresh installed Windows 11 Pro).
```
PS C:\Users\yhabteab\Workspace\icinga2\build> .\Bin\Release\Debug\icinga2.exe --version
icinga2.exe - The Icinga 2 network monitoring daemon (version: v2.14.0-94-g88a32cdd0; debug)

Copyright (c) 2012-2023 Icinga GmbH (https://icinga.com/)
License GPLv2+: GNU GPL version 2 or later <https://gnu.org/licenses/gpl2.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

System information:
  Platform: Windows
  Platform version: Windows 10 Pro 23H2
  Kernel: Windows
  Kernel version: 6.3.22631
  Architecture: x86_64

Build information:
  Compiler: MSVC 19.29.30153.0                                                                                            Build host: YONASHABTEAF0EE                                                                                             OpenSSL version: OpenSSL 3.0.12 24 Oct 2023
```

Reply:
It's already better than the master branch though:
```bash
System information:
  Platform: Windows
  Platform version: 8
  Kernel: Windows
  Kernel version: 6.2
  Architecture: x86_64
```